File: programmers/코딩테스트 고득점 Kit/정렬/가장 큰 수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cstring>
#include<iostream>
using namespace std;
bool compare(string a, string b) {
	string ab = "", ba = "";
	ab += a, ab += b;
	ba += b, ba += a;
	return stoi(ab) > stoi(ba);
}
string solution(vector<int> numbers) {
	string answer = "";
	vector<string> str;
	for (int i = 0; i < numbers.size(); i++)
		str.push_back(to_string(numbers[i]));
	sort(str.begin(), str.end(), compare);
	for (int i = 0; i < str.size(); i++) {
		if (answer == "" && str[i] == "0") continue;
		answer += str[i];
	}
	if (answer == "") answer = "0";
	return answer;
}

File: programmers/코딩테스트 고득점 Kit/정렬/가장 큰 수.java
import java.util.*;
import java.util.stream.Collectors;
class Solution {
    public String solution(int[] numbers) {
        StringBuilder answer= new StringBuilder();
        List<String> arr = new ArrayList<>();
        for (int number : numbers) {
            arr.add(String.valueOf(number));
        }
        arr = arr.stream().sorted((a, b) -> {
            String ab = a+b;
            String ba = b+a;
            Integer abNum = Integer.valueOf(ab);
            Integer baNum = Integer.valueOf(ba);
            return baNum - abNum;
        }).collect(Collectors.toList());
        for (String s : arr) {
            if (answer.toString().equals("") && s.equals("0")) continue;
            answer.append(s);
        }
        if(answer.toString().equals("")) return "0";
        return answer.toString();
    }
}